src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,102 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    static final float DEFAULT_LOADER_SIZE = 0.75f;
+    static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private int size;
+    private int threshold;
+    private int capacity;
+    private Node<K, V> [] bucket;
+
+    MyHashMap() {
+        bucket = new Node[DEFAULT_INITIAL_CAPACITY];
+        threshold = (int) (DEFAULT_INITIAL_CAPACITY * DEFAULT_LOADER_SIZE);
+        capacity = DEFAULT_INITIAL_CAPACITY;
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size + 1 > threshold) {
```

Comment:
```suggestion
        if (size  > threshold) {
```

Reply:
from my point of view, its better to check if we need ti resize begore adding another element

Reply:
you are trying to resize you array when size == threshold, but you should do it only when size > threshold.
size = 12, threshold = 12 and you are already resizing, thats not good